All_Relationships/6.true_relationship_preparation.py

```diff
@@ -0,0 +1,243 @@
+
+    # Yates continuity correction for confidence interval of delta
+    correction = 0.5*(1/n1 + 1/n2)
+    corrected_ci = (ci[0] - correction, ci[1] + correction)
+
+    return delta, ci, corrected_ci
+
+
+# In[7]:
+
+
+total = candidate_df["sentence_count"].sum()
+odds = []
+p_val = []
+expected = []
+lower_ci = []
+
+for disease, gene in tqdm.tqdm(zip(candidate_df["disease_id"],candidate_df["gene_id"])):
+    cond = (candidate_df["disease_id"] == disease) & (candidate_df["gene_id"] == gene)
+    a = candidate_df[cond]["sentence_count"].values[0] + 1
+                                        
+    cond = (candidate_df["disease_id"] != disease) & (candidate_df["gene_id"] == gene)
+    b = sum(candidate_df[cond]["sentence_count"].values) + 1
+    
+    cond = (candidate_df["disease_id"] == disease) & (candidate_df["gene_id"] != gene)
+    c = sum(candidate_df[cond]["sentence_count"].values) +1
+    
+    cond = (candidate_df["disease_id"] != disease) & (candidate_df["gene_id"] != gene)
+    d = sum(candidate_df[cond]["sentence_count"].values) + 1
+    
+    c_table = np.array([[a, b], [c, d]])
+    
+    # Gather confidence interval
+    delta, ci, corrected_ci = diffprop(c_table)
+    lower_ci.append(ci[0])
+    
+    # Gather corrected odds ratio and p_values
+    odds_ratio, p_value = fisher_exact(c_table, alternative='greater')
+    odds.append(odds_ratio)
+    p_val.append(p_value)
+    
+    
+    total_disease = candidate_df[candidate_df["disease_id"] == disease]["sentence_count"].sum()
+    total_gene = candidate_df[candidate_df["gene_id"] == gene]["sentence_count"].sum()
+    expected.append((total_gene * total_disease)/float(total))
+
+candidate_df["p_value"] = p_val
+candidate_df["odds_ratio"] = odds
+candidate_df["expected_sen"] = expected
+candidate_df["lower_ci"] = lower_ci
+
+
+# In[ ]:
+
+
+candidate_df.sort_values("p_value", ascending=True).head(20)
+
+
+# # Combine sentence marginals
+
+# In[19]:
+
+
+train_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_train_marginals.csv")
+dev_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_dev_marginals.csv")
+test_marginals_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/lstm_test_marginals.csv")
+
+
+# In[20]:
+
+
+train_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/train_candidates_sentences.csv")
+dev_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/dev_candidates_sentences.csv")
+test_sentences_df = pd.read_csv("stratified_data/lstm_disease_gene_holdout/test_candidates_sentences.csv")
+
+
+# In[23]:
+
+
+train_sentences_df["marginals"] = train_marginals_df["RNN_marginals"].values
+dev_sentences_df["marginals"] = dev_marginals_df["RNN_marginals"].values
+test_sentences_df["marginals"] = test_marginals_df["RNN_10_Marginals"].values
+
+
+# # Combine Marginal Probabilities
+
+# In[30]:
+
+
+candidate_df = pd.read_csv("disease_gene_summary_stats.csv")
```

Comment:
It is -1 because snorkeling's encoding is 1 for true, 0 for unknown and -1 for false. Could change it to 0/1, but I think it will be okay with -1 and 1.